src/parse.c

```diff
@@ -10480,15 +10533,118 @@ static tree_t p_psl_clock_declaration(void)
    psl_node_t p = psl_new(P_CLOCK_DECL);
    psl_set_tree(p, expr);
 
+   tree_set_ident(parent, well_known(W_DEFAULT_CLOCK));
    consume(tSEMI);
 
    psl_set_loc(p, CURRENT_LOC);
    psl_check(p);
 
+   return p;
+}
+
+static psl_node_t p_psl_property_declaration(tree_t parent)
+{
+   // property PSL_Identifier [ ( Formal_Parameter_List ) ] is Property ;
+
+   assert(tree_kind(parent) == T_PSL);
+
+   BEGIN("PSL property declaration");
+
+   scan_as_psl();
+
+   psl_node_t decl = psl_new(P_PROPERTY_DECL);
+   consume(tPROPERTY);
+
+   scan_as_vhdl();
+   tree_set_ident(parent, p_identifier());
+   scan_as_psl();
```

Comment:
```suggestion
   tree_set_ident(parent, p_identifier());
```

Switching the mode here shouldn't be necessary as `tID` will be returned in either.